### DISCUSSION TITLE
Flutter macOS app failed publishing to App Store Connect

### DISCUSSION THREAD

**Discussion by korenminchev** [codemagic-ci-cd/codemagic-docs#1498] [Q&A] [answered]
I have a Flutter app that successfully publishes when building for ios.
When I try to publish the macOS version I get the following error:
`*** Error: ERROR: Asset validation failed (90255) The installer package includes files that are only readable by the root user. This will prevent verification of the application's code signature when your app is run. Ensure that non-root users can read the files in your app.`

I can also publish it successfully when building from my mac.

I suspect this happens because CodeMagic builds the app with root privileges. Any way to disable that?
Thank you!

**Comment by icarusdust** [comment, marked as answer]
Hi @korenminchev, looks like it is about the incorrect *.frag shader file permissions: https://github.com/flutter/flutter/issues/108737
In order to tackle it, you can either in Xcode custom script or in Codemagic post-build (after the build command is complete) script can add the following bits:

```
chmod 0644 "$TARGET_BUILD_DIR/$FULL_PRODUCT_NAME/Contents/Frameworks/App.framework/Versions/A/Resources/flutter_assets/shaders/ink_sparkle.frag"
chmod 0644 "$TARGET_BUILD_DIR/App.framework/Versions/A/Resources/flutter_assets/shaders/ink_sparkle.frag"
```

Let me know if it helps.

**Reply by korenminchev** [reply]
Hey @icarusdust, I added the commands but got the same error. I also looked at the flutter issue to validate.
<img width="1027" alt="image" src="https://user-images.githubusercontent.com/26023554/226706523-8dc12817-87b0-40e3-9e24-983e41f5f72c.png">
I don't know if you have access to builds, but here's the build id if you want to take a look: `6419f25fabf5e2fe31bb6f54`

**Reply by icarusdust** [reply]
Try it in Xcode as explained here: https://github.com/flutter/flutter/issues/111808#issuecomment-1264558589

Also, I see that your team is on a paid plan which means that you have access to the private chat widget in the UI. As responses are faster in private chat for paying customers, you should reach out using that widget next time.

**Reply by korenminchev** [reply]
Thank you!
